java/algorithm/src/test/java/org/apache/arrow/algorithm/sort/TestInsertionSorter.java

```diff
@@ -43,51 +45,49 @@ public void shutdown() {
     allocator.close();
   }
 
-  private void testSortIntVectorRange(int start, int end, int vectorLength) {
+  private static final int VECTOR_LENGTH = 10;
+
+  private void testSortIntVectorRange(int start, int end, int[] expected) {
     try (IntVector vector = new IntVector("vector", allocator);
          IntVector buffer = new IntVector("buffer", allocator)) {
 
-      vector.allocateNew(vectorLength);
       buffer.allocateNew(1);
 
-      for (int i = 0; i < vectorLength; i++) {
-        // place values in reverse order
-        vector.set(i, vectorLength - i - 1);
-      }
+      ValueVectorDataPopulator.setVector(vector, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0);
+      assertEquals(VECTOR_LENGTH, vector.getValueCount());
 
       VectorValueComparator<IntVector> comparator =
           DefaultVectorComparators.createDefaultComparator(vector);
       InsertionSorter.insertionSort(vector, start, end, comparator, buffer);
 
-      // verify results
-      for (int i = start, expected = vectorLength - end - 1; i < end; i++, expected++) {
-        assertEquals(expected, vector.get(i));
+      assertEquals(VECTOR_LENGTH, expected.length);
+      for (int i = 0; i < VECTOR_LENGTH; i++) {
+        assertFalse(vector.isNull(i));
+        assertEquals(expected[i], vector.get(i));
       }
     }
   }
 
   @Test
   public void testSortIntVector() {
-    testSortIntVectorRange(0, 99, 100);
-    testSortIntVectorRange(15, 30, 100);
-    testSortIntVectorRange(50, 50, 100);
-    testSortIntVectorRange(50, 51, 100);
-    testSortIntVectorRange(0, 30, 100);
-    testSortIntVectorRange(72, 99, 100);
+    testSortIntVectorRange(2, 5, new int[] {9, 8, 4, 5, 6, 7, 3, 2, 1, 0});
+    testSortIntVectorRange(3, 7, new int[] {9, 8, 7, 2, 3, 4, 5, 6, 1, 0});
+    testSortIntVectorRange(3,4, new int[] {9, 8, 7, 5, 6, 4, 3, 2, 1, 0});
```

Comment:
nit: please fix spacing here and below.

Reply:
Resolved. Thanks for your careful review.